### PR TITLE
Convert s2ir.c to D

### PR DESCRIPTION
**Pull request by WalterBright**
Initiate work on this.

### CODE REVIEW

**Comment by WalterBright** [comment]
Ready to go. I don't know why github won't show the changes as a diff, oh well, but s2ir.c and s2ir.d are line-by-line comparable.

**Comment by UplinkCoder** [comment]
Awesome!

File: src/backend/rtlsym.h [hunk 1]
```diff
@@ -19,13 +19,13 @@
         tsdlib          TYjfunc, C mangling
  */
 
-#if SCPP
+#if 1 || SCPP
```
Comment (andralex) [inline src/backend/rtlsym.h]: Can you please make a subsequent pass minding these obsoleted conditionals?

**Review by andralex — APPROVED** [review]
+1932 -1752

Wonder why so many additions.

**Comment by andralex** [comment]
Auto-merge toggled on

**Comment by MartinNowak** [comment]
Why are you guys ignoring tests? You just broke 3 platforms and the nightly builds.
https://travis-ci.org/dlang/dmd/builds/167844367

**Comment by andralex** [comment]
My bad, apologies. Will pay more attention to Travis in the future.

**Comment by MartinNowak** [comment]
> My bad, apologies. Will pay more attention to Travis in the future.

Let's hope it's soon stable enough to make it required as well.

**Comment by UplinkCoder** [comment]
Hmm the auto-tester did not complain on those platforms.
It seems only to occur with ldc

**Comment by MartinNowak** [comment]
Yes, the Travis-CI tests are complementary to the auto-tester. They test building dmd with other host compilers and self-compilation.
The OSX tests on Travis-CI we're added to cover clang and host C++ compiler.

**Pull request merged**